### PR TITLE
Fix "Enable fallback if binary name given" bugs

### LINKED ISSUE #5280: "Enable fallback if binary name given" bugs
While the idea behind #5270 is reasonable, the implementation we merged here needs more work. As it is, I think it introduced bugs.

In `x86-sse.h`, we have:

```c
#ifndef CPU_FALLBACK_BINARY_DEFAULT
#define CPU_FALLBACK_BINARY_DEFAULT     "john-non-sse"
#endif

#ifndef CPU_FALLBACK
#define CPU_FALLBACK                    0
#endif
#if !defined(CPU_FALLBACK_BINARY) && defined(CPU_FALLBACK_BINARY_DEFAULT)
#define CPU_FALLBACK_BINARY             CPU_FALLBACK_BINARY_DEFAULT
#endif
```

In `x86-mmx.h`, we have:

```c
#ifndef CPU_FALLBACK
#define CPU_FALLBACK                    0
#endif
#ifndef CPU_FALLBACK_BINARY
#define CPU_FALLBACK_BINARY             "john-non-mmx"
#endif
```

So building with these now, we'll end up with fallback enabled where that wasn't intended. We probably need to revise these.

Also, while the old behavior is documented in a comment in `params.h` and in some files under `doc/`, the new behavior is undocumented. Maybe that's fine as documenting both the old and the new behavior would be cryptic, and the old behavior is also still relevant when default filenames are used and not overridden.

### PR DESCRIPTION
**Pull request by solardiz**
Fixes #5280

### CODE REVIEW

File: src/x86-mmx.h [hunk 1]
```diff
@@ -46,7 +46,7 @@
 #ifndef CPU_FALLBACK
 #define CPU_FALLBACK			0
 #endif
-#ifndef CPU_FALLBACK_BINARY
+#if CPU_FALLBACK && !defined(CPU_FALLBACK_BINARY)
```
Comment (ghost) [inline src/x86-mmx.h:49]: LGTM.

File: src/Makefile.legacy [hunk 2]
```diff
@@ -573,7 +573,7 @@ linux-x86-64-32-any:
 		LDFLAGS="$(LDFLAGS) -m32 -lcrypt -ldl" \
 		AESNI_ARCH=86 YASM_FORMAT="elf32"
 	@echo "Failing after this point just means some helper tools did not build:"
-	$(MAKE_ORIG) $(PROJ_PCAP) CFLAGS="$(CFLAGS) -m32"
+	$(MAKE_ORIG) $(PROJ_PCAP) CFLAGS="$(CFLAGS) -m32" LDFLAGS="$(LDFLAGS) -m32"
```
Comment (ghost) [inline src/Makefile.legacy:576]: Isn't this a hint that #5240 '-m64' is needed, at least in the meantime?

**Review by ghost — APPROVED** [review]
I think I cannot test this *-32 bits builds on recent Ubuntu. Anyway, looks like good stuff.

File: src/Makefile.legacy (hunk 2 above)
Comment (solardiz) [reply src/Makefile.legacy:576]: No, needing to use `-m32` to build/link 32-bit code on a typical 64-bit system is a hint that such system actually _doesn't_ need explicit `-m64` to build/link 64-bit code.

**Pull request merged**